### PR TITLE
doc: Explain in the generate* RPC docs that they are only for testing

### LINKED ISSUE #24027: RPC generateblock doesn't mine a block
Here's what the Bitcoin generateblock RPC  documentation says.

_Mine a block with a set of ordered transactions immediately to a specified address or descriptor (before the RPC call returns)_

_Argument #1
The address or descriptor to send the newly generated bitcoin to._

https://developer.bitcoin.org/reference/rpc/generateblock.html

I deployed a full node, downloaded the entire Blockchain and sent a generateblock request with 1 transaction from getrawmempool to avoid sorting issues.

The generic error was "Failed to make a block" and the Bitcoin wasn't sent.

I also made a video of the bug.

https://www.tiktok.com/@ghostcompute/video/7051701541996268846?sender_device=mobile&sender_web_id=7042509351429473798&is_from_webapp=v1&is_copy_url=0

### PR DESCRIPTION
**Pull request by sipa**

### CODE REVIEW

**Comment by fjahr** [comment]
ACK e187e0ef6e0cafa6236aa26f7b033d21264bb044

**Review by w0xlt — APPROVED** [review]
ACK e187e0e

**Review by ghost — APPROVED** [review]
ACK https://github.com/bitcoin/bitcoin/pull/24028/commits/e187e0ef6e0cafa6236aa26f7b033d21264bb044

This PR improves documentation for `generateblock` `generatetoaddress` `generatetodescriptor` RPC and fixes https://github.com/bitcoin/bitcoin/issues/24027

File: src/rpc/mining.cpp [hunk 1]
```diff
@@ -210,7 +210,10 @@ static RPCHelpMan generatetodescriptor()
 {
     return RPCHelpMan{
         "generatetodescriptor",
-        "\nMine blocks immediately to a specified descriptor (before the RPC call returns)\n",
+        "\nMine blocks immediately to a specified descriptor (before the RPC call returns)\n"
```
Comment (MarcoFalke) [inline src/rpc/mining.cpp:213]:
```suggestion
        "Mine blocks immediately to a specified descriptor (before the RPC call returns)\n"
```

While touching this, there is no need to keep the `\n`s in the beginning and end.

**Review by maflcko — COMMENTED** [review]
No objection to updating the docs, but I think the real issue is that the docs aren't hidden. For some reason no one ever removed them from the rpc `help` overview. I'd suggest to do that instead.


And once the RPCs are hidden, I think we can assume that anyone reading their help text understand POW. So a very brief "testing only because ridiculously inefficient" is enough.

File: src/rpc/mining.cpp (hunk 1 above)
Comment (jonatack) [reply src/rpc/mining.cpp:213]: suggestions for each of these helps
- period at end of sentences
- shorten the longer line
- maybe s/maxtries/`maxtries`/
- omit newlines at start and end
 
```diff
-        "\nMine blocks immediately to a specified descriptor (before the RPC call returns)\n"
-        "Only maxtries hash attempts will be performed. If no valid block can be found among those,\n"
-        "an error will be returned. This RPC is only intended for testing purposes, as it is highly inefficient and extremely\n"
-        "unlikely to succeed on mainnet\n",
+        "Mine blocks immediately to a specified descriptor (before the RPC call returns).\n"
+        "Only `maxtries` hash attempts will be performed. If no valid block can be found among those,\n"
+        "an error will be returned. This RPC is only intended for testing purposes, as it is highly\n"
+        "inefficient and extremely unlikely to succeed on mainnet.",
```
output
```
Mine blocks immediately to a specified descriptor (before the RPC call returns).
Only `maxtries` hash attempts will be performed. If no valid block can be found among those,
an error will be returned. This RPC is only intended for testing purposes, as it is highly
inefficient and extremely unlikely to succeed on mainnet.
```

Comment (MarcoFalke) [reply src/rpc/mining.cpp:213]: If this turns into a pull to rework the docs, the "immediately" and "(before the RPC call returns)" can be removed. It should be clear from the return value (the block hash) that the block was generated before the has was returned.

**Comment by laanwj** [comment]
Concept ACK

**Comment by theStack** [comment]
Concept ACK

File: src/rpc/mining.cpp [hunk 2]
```diff
@@ -210,7 +210,10 @@ static RPCHelpMan generatetodescriptor()
...
+        "Only maxtries hash attempts will be performed. If no valid block can be found among those,\n"
+        "an error will be returned. This RPC is only intended for testing purposes, as it is highly inefficient and extremely\n"
```
Comment (MarcoFalke) [inline src/rpc/mining.cpp:215]: nit: This is wrong

```suggestion
        "an empty array will be returned. This RPC is only intended for testing purposes, as it is highly inefficient and extremely\n"
```

**Review by maflcko — COMMENTED** [review]
The docs in this pull aren't exactly right (no error will be returned). I've created a conflicting pull in #24137 due to inactivity here. Feel free to cherry-pick or rebase on that one.

**Comment by MarcoFalke** [comment]
@sipa Are you still working on this?

**Comment by MarcoFalke** [comment]
Closing for now, but this can be reopened anytime (or a new pull can be created).

**Pull request closed without merging**